src/edframe.cpp

```diff
@@ -962,7 +962,7 @@ bool PoeditFrame::NeedsToAskIfCanDiscardCurrentDoc() const
 }
 
 template<typename TFunctor>
-void PoeditFrame::DoIfCanDiscardCurrentDoc(TFunctor completionHandler)
+void PoeditFrame::DoIfCanDiscardCurrentDoc(TFunctor completionHandler, bool bSkipUpload)
```

Comment:
Sorry, but this is an obvious hack working *around* whatever the issue is (this is unclear to me without repro steps) instead of *fixing* it where it was caused.

Can you please provide steps to reproduce so that I can better understand what is this fixing?

Reply:
Did you see [this my comment/answer and commit mentioned there](https://github.com/vslavik/poedit/pull/633#discussion_r439735130)?  And paricularly [this comment in the code ](https://github.com/vslavik/poedit/commit/508e4f920401ff39466007adf66754b730118123#diff-52c8dd65648ccef63c74b9aa24e24da6R1771). Sorry if it's still just not enough clear/obvious to reproduce by so more detailed step-by-step is:
1. In the released `v2.3` Poedit open any `PO` file from Crowdin (via `Open from Crowdin` or just downloaded in browser via Crowdin UI)
2. Make any changes to any translation in that file in Poedit
3. Press `Sync`
4. Press `Yes``Save file before sync`
    - Result: If you put breakpoint or log entry in `CrowdinClient::UploadFile()` - you will notice it's called twice on this step (i.e. after above `OK`/`Yes`

Reply:
I created an issue for this (bSkipUpload) as there seems to be more going on (race, bad UI) — see #638